### DISCUSSION TITLE
Process of Adding New Search Parameters?

### DISCUSSION THREAD

**Discussion by KMCGit** [FirelyTeam/spark#516] [Q&A] [answered]
What would be the process for adding New/Custom Search Parameters? 
(Thanks in advance for your time and assistance)

**Comment by kennethmyhra** [comment, marked as answer]
In Spark you can do this in code in the `ConfigureServices()` of your `Startup.cs file`. Use the extension method `AddCustomSearchParameters`: https://github.com/FirelyTeam/spark/blob/02125beb85f5455450a391cbc5b57aeb7b06c48a/src/Spark.Engine/Extensions/NetCore/IServiceCollectionExtensions.cs#L258

There is a simpler example using Spark's NuGet packages here:  https://github.com/incendilabs/spark-example. You can add the `services.AddCustomSearchParameters(...)` statement below the `AddFhir()` statement here: https://github.com/incendilabs/spark-example/blob/2a7c9edc2bfbd74dd65ebe7b5ec5c80a5f3cf127/Startup.cs#L35-L51.
